docs/developer-guide.md

```diff
@@ -1 +1,91 @@
 # Developer Guide
+
+This guide will help you get started with developing, deploying, and running Cube AI.
+
+## Cloning the Repository
+
+```bash
+https://github.com/ultravioletrs/cube.git
+cd cube
+```
+
+## Pulling Docker Images
+
+```bash
+cd cube/docker/
+docker compose pull
+```
+
+## Running Services with Docker Compose
+
+You can run/start Cube AI services using Docker Compose as described in [this guide](https://github.com/ultravioletrs/cube/blob/main/hal/ubuntu/README.md).
+
+To properly access Cube AI UI deployed on a different server, update the IP address entries in `docker/.env` as described in the above guide to point to your server IP address. The Cube AI UI can then be accessed through your browser at:
+
+```bash
+http://<your-server-ip-address>:3001
+```
+
+For example, if you have deployed locally, use:
+
+```bash
+http://localhost:3001
+```
+
+## Building Docker Images
+
+You can build the Docker images for Cube AI and related services using the following commands in the project's root directory.
+
+To build the Cube Proxy image, use the `docker/Dockerfile` provided in the project.
+
+```bash
+docker build -t ghcr.io/ultravioletrs/cube/proxy:latest -f docker/Dockerfile .
+```
+
+For development builds, use the `docker/Dockerfile.dev` provided in the project.
+
+```bash
+docker build -t cube-proxy:dev -f docker/Dockerfile.dev .
+```
+
+If you prefer using `make`, the `Makefile` in the project includes a command to build the latest Docker image:
+
+```bash
+make latest
```

Comment:
This is incorrect. `latest` builds the docker prod image and pushes it to the docker registry. Change this